File: PROGRAMMERS/올바른괄호.py
from collections import deque
def solution(s):
    answer = True
    d = deque()
    
    for si in s:
        if si == '(':
            d.append(si)
            continue
        if len(d) > 0 and d[-1] == '(':
            d.pop()
        else:
            return False
    if len(d) > 0:
        return False
    return True

File: PROGRAMMERS/올바른괄호.cpp
#include<string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s)
{
    bool answer = true;

    stack<char> st;
    for(int i=0;i<s.size();i++){
        if(s[i] == '(') st.push('(');
        else{
            if(st.empty() || st.top() != '(') return false;
            else st.pop();
        }
    }
    if(st.size() > 0) answer = false;
    return answer;
}